conans/test/functional/workspace/cmake_workspace_test.py

```diff
@@ -0,0 +1,242 @@
+        t.run('install conanfile.txt -g virtualrunenv')
+        executable = self.libG.executables[0]
+        if platform.system() != "Windows":
+            t.run_command("bash -c 'source activate_run.sh && {}'".format(executable.name))
+        else:
+            t.run_command("activate_run.bat && {}.exe".format(executable.name))
+        self.assertMultiLineEqual(str(t.out).strip(), self.original_output.strip())
+
+    def test_created_projects(self):
+        self.run_outside_ws()
+
+    def test_modify_editable(self):
+        """ If an editable package is modified, then only the changes into CPP files
+            will be reflected because the linking is done in the root project (given
+            ABI compatibility).
+        """
+        t = TestClient(base_folder=self.base_folder)
+        ws_yml = Template(workspace_yml_template).render(editables=self.editables)
+        t.save({'ws.yml': ws_yml}, clean_first=True)
+        t.run("workspace install ws.yml")
+
+        t.run_command('mkdir build')
+        t.run_command('cd build && cmake .. -DCMAKE_MODULE_PATH="{}"'.format(t.current_folder))
+        t.run_command('cd build && cmake --build .')
+        self.assertIn("Built target {}".format(self.libC.name), t.out)
+        self.assertIn("Built target {}".format(self.libF.name), t.out)
+        t.run_command('cd build && ./bin/pkgG_exe')
+        self.assertMultiLineEqual(str(t.out).strip(), self.original_output.strip())
+
+        self.libA.modify_cpp_message("Edited!!!")  # It will change value in cpp and in header file
+        t.run_command('cd build && cmake --build .')
+        self.assertIn("Built target {}".format(self.libC.name), t.out)
+        self.assertIn("Built target {}".format(self.libF.name), t.out)
+        t.run_command('cd build && ./bin/pkgG_exe')
+
+        # Check the resulting message with the 'Editted!!!'
+        original_output = self.original_output.replace("> pkgA: default",
+                                                       "> pkgA: Edited!!!")
+        # TODO: FIXME, if pkgC is compiled, then all pkgA_header would have the Edited!!! message.
+        original_output = original_output.replace(
+            "\t\t> pkgB: default\n\t\t\t> pkgA_header: default",
+            "\t\t> pkgB: default\n\t\t\t> pkgA_header: Edited!!!")
+        self.assertMultiLineEqual(str(t.out).strip(), original_output)
+
+        # And outside the workspace, everything should behave exactly the same
+        self.run_outside_ws()
+
+    def test_editables_shared(self):
+        """ All the editable packages are built as shared libs """
+        t = TestClient(base_folder=self.base_folder)
+        ws_yml = Template(workspace_yml_template).render(editables=self.editables)
+        t.save({'ws.yml': ws_yml}, clean_first=True)
+
+        t.run("workspace install ws.yml")
+        t.run_command('mkdir build')
+        t.run_command('cd build && cmake ..'
```

Comment:
Having seen all these flags, it comes to my mind the idea of creating a `conanfile.py` to use build helpers to build/handle the workspace... 🤔